### PR TITLE
[DOCS] Document that doc_as_upsert does not support ingest pipelines

### LINKED ISSUE #30724: Document ingest limitations
Ingest does not process update requests and support the `doc_as_upsert` option properly. This should be clearly documented.

Relates to #30627

### PR DESCRIPTION
**Pull request by asfalots**
Closes #30724 

Just a small documentation update to avoid losing time.

### CODE REVIEW

**Comment by jrodewig** [comment]
@asfalots Can you sign our CLA using the same email address associated with your commit:

https://www.elastic.co/contributor-agreement

You can find the email address here: 
https://github.com/elastic/elasticsearch/commit/3f523e152debfdbba01b48e310e509ec26c0d7d5.patch

We can't merge pull requests without a signed CLA.

**Comment by asfalots** [comment]
> @asfalots Can you sign our CLA using the same email address associated with your commit:

Sorry, I used my own address and not the github one.  Fixed now.

**Comment by jrodewig** [comment]
@elasticmachine test this please

File: docs/reference/docs/update.asciidoc [hunk 1]
```diff
@@ -333,3 +333,7 @@ POST test/_update/1
 }
 --------------------------------------------------
 // TEST[continued]
+[NOTE]
+====
+Using pipelines with `doc_as_upsert` is not possible.
```
Comment (jrodewig) [inline docs/reference/docs/update.asciidoc:338]:
```suggestion
Using <<ingest,ingest pipelines>> with `doc_as_upsert` is not supported.
```

**Review by jrodewig — APPROVED** [review]
LGTM. Thanks @asfalots

**Comment by jrodewig** [comment]
@elasticmachine test this please

**Comment by jrodewig** [comment]
### Backport commits
master 853dd1b 
7.x 6c93fed 
7.8 c92e1f7
7.7 a679f5c

**Comment by puppylpg** [comment]
i think the doc is put on the wrong page. `update` doesn't even support `pipeline` parameter, so why writing this sentence in the update api page? 
> Using [ingest pipelines](https://www.elastic.co/guide/en/elasticsearch/reference/current/ingest.html) with doc_as_upsert is not supported.

AFAIK, https://github.com/elastic/elasticsearch/issues/30627 means that `pipeline` is not supported in `bulk` api when `doc_as_upsert` is used. So the doc above ought to be added onto bulk api page, right?

Or maybe i misunderstood something, and will be sorry to interrupt...

**Pull request merged**